src/main/java/org/mskcc/cmo/metadb/service/impl/UpdateMessageHandlingServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+                LOG.info("Received message on topic: " + VALIDATOR_REQUEST_UPDATE_TOPIC);
+                try {
+                    String requestJson = mapper.readValue(
+                            new String(msg.getData(), StandardCharsets.UTF_8),
+                            String.class);
+                    updateMessageHandlingService.requestUpdateFilterHandler(requestJson);
+                } catch (Exception e) {
+                    LOG.error("Exception during processing of request on topic: "
+                            + VALIDATOR_REQUEST_UPDATE_TOPIC, e);
+                    // TODO: should set ups a request status logger here
+                }
+            }
+        });
+    }
+
+    private void setupSampleUpdateFilterHandler(Gateway gateway,
+            UpdateMessageHandlingService updateMessageHandlingService)
+            throws Exception {
+        gateway.subscribe(VALIDATOR_SAMPLE_UPDATE_TOPIC, Object.class, new MessageConsumer() {
+            public void onMessage(Message msg, Object message) {
+                LOG.info("Received message on topic: " + VALIDATOR_SAMPLE_UPDATE_TOPIC);
+                try {
+                    String sampleJson = mapper.readValue(
+                            new String(msg.getData(), StandardCharsets.UTF_8),
+                            String.class);
+                    updateMessageHandlingService.sampleUpdateFilterHandler(sampleJson);
+                } catch (Exception e) {
+                    LOG.error("Exception during processing of sample on topic: "
+                            + VALIDATOR_SAMPLE_UPDATE_TOPIC, e);
+                }
+            }
+        });
+    }
+
+    @Override
+    public void shutdown() throws Exception {
+        if (!initialized) {
+            throw new IllegalStateException("Message Handling Service has not been initialized");
+        }
+        exec.shutdownNow();
+        requestUpdateFilterHandlerShutdownLatch.await();
+        sampleUpdateFilterHandlerShutdownLatch.await();
+        shutdownInitiated = true;
+    }
+
+    private Boolean isCmoSample(String sampleJson) throws JsonProcessingException {
+        Map<String, Object> sampleJsonMap = mapper.readValue(sampleJson, Map.class);
+        if (sampleJsonMap.get("isCmoSample") == null
+                || Strings.isBlank(sampleJsonMap.get("isCmoSample").toString())) {
+            return Boolean.FALSE;
+        }
+        return Boolean.valueOf(sampleJsonMap.get("isCmoSample").toString());
+    }
+
+    private Boolean hasRequestId(String sampleJson) throws JsonProcessingException {
+        Map<String, Object> sampleJsonMap = mapper.readValue(sampleJson, Map.class);
+        if ((sampleJsonMap.get("igoRequestId") == null
+                || Strings.isBlank(sampleJsonMap.get("igoRequestId").toString()))) {
+            return Boolean.FALSE;
+        }
+        return Boolean.valueOf(sampleJsonMap.get("igoRequestId").toString());
```

Comment:
[requested change] some of this overlaps with the helper methods in the other message handler class:

suggestions to consider:
- consolidate into some utility class that can be used in both message handlers
- and/or keep everything in one message handler